### PR TITLE
Publish short-form posts as native Bluesky posts (post-format-aware composition)

### PR DESCRIPTION
**Pull request by kraftbj**
## Context

Proposed as part of the FOSSE plugin's design plan for native Bluesky publishing — the design lives at [`Automattic/fosse#18`](https://github.com/Automattic/fosse/pull/18) (see `sdd/bluesky-native-publishing/`). The implementation here is a faithful expression of that plan.

- **If you disagree with WHAT this does** — the discriminator logic (mirrors AP's `get_type()`), the filter name, the short-form composition (body-as-text, no embed), or the small refactor — the FOSSE SDD is the right place to push back. Happy to carry feedback in either direction.
- **Code-level concerns** (style, edge cases, test coverage, whether the refactor should be split) are perfect for here.

Sibling PR adding `activitypub_post_object_type` to the AP plugin: [Automattic/wordpress-activitypub#3210](https://github.com/Automattic/wordpress-activitypub/pull/3210). A small FOSSE projector hooks both filters so a single user-level "always Note" preference projects onto both networks.

## Summary

Currently, every published post federates to Bluesky as `title + excerpt + permalink` truncated to 300 graphemes, with an `app.bsky.embed.external` link card pointing back to WordPress. That's correct for long-form blog posts — readers want a card, not a chopped-up essay in their timeline. But it's the wrong shape for a short post: the note *is* the content, and a link card back to a page that repeats those 50 characters is noise.

This PR makes the `app.bsky.feed.post` composition post-format-aware, mirroring the ActivityPub plugin's [`Post::get_type()`](https://github.com/Automattic/wordpress-activitypub/blob/trunk/includes/transformer/class-post.php) discriminator:

- **Short-form** (post has no title, OR has any non-empty `post_format`): body becomes the Bluesky text directly, defensively clamped at 300 graphemes; no external embed.
- **Long-form** (titled with no format): existing behavior unchanged — title + excerpt + permalink teaser with external card.

The change is symmetric with what the AP plugin already does for Mastodon (`Note` vs. `Article`), so a user who sets `post_format = status` on a post gets consistent short-form treatment on both networks without any extra setup.

## Changes

- `includes/transformer/class-post.php`
  - New private `is_short_form( WP_Post $post ): bool` mirroring AP's `get_type()`.
  - New private `build_short_form_text(): string` — rendered plain-text body, clamped to 300 via `truncate_text`.
  - `transform()` branches on the new `atmosphere_is_short_form_post` filter (default: the `is_short_form()` computed value). Short-form → body-as-text + null embed. Long-form → existing `build_text()` / `build_embed()` path unchanged.
- `includes/transformer/class-base.php` — new protected `render_post_content_plain( WP_Post $post ): string` extracting the plain-text rendering pattern currently inlined in `Document::get_text_content()`.
- `includes/transformer/class-document.php` — `get_text_content()` now delegates to the shared helper. Pure refactor, no behavior change.
- `tests/phpunit/tests/transformer/class-test-post.php` (new) — 8 tests covering long-form unchanged, all short-form triggers (untitled, status format, aside format), over-cap truncation, filter force short/long, filter receives the right arguments.
- `.github/changelog/add-short-form-post-path` — minor/added entry.

## Why a new filter and not a record-level rewrite

`atmosphere_transform_bsky_post` exists and could technically be used to rewrite the composed record, but it would force consumers to duplicate the short-form composition logic and re-extract facets against the new text. A yes/no discriminator filter (`atmosphere_is_short_form_post`) is the smaller extension point and composes naturally with the existing `atmosphere_transform_bsky_post` final-adjustment filter.

## Test plan

- [x] PHP syntax clean on all four modified/new files.
- [x] `composer lint` clean on all modified files.
- [ ] `composer test` / `npm run env-test` — I did not run locally (no MySQL / wp-env set up in my environment). Confident via reading + syntax check; please let me know if CI surfaces anything.

## Note on the refactor

The `Document::get_text_content()` → `Base::render_post_content_plain()` extraction is bundled here because the new short-form path needs the same rendering and it'd be strange to have two copies. Happy to split it into its own prep PR if you'd prefer that shape — let me know.

Fixes DOTCOM-16838

### CODE REVIEW

File: includes/transformer/class-post.php [hunk 1]
```diff
@@ -49,7 +49,32 @@ class Post extends Base {
...
+			$this->object
+		);
+
+		if ( $is_short ) {
+			$text  = $this->build_short_form_text();
+			$embed = null;
+		} else {
+			$text  = $this->build_text();
+			$embed = $this->build_embed();
+		}
```
[the remark this answers is not in the document]
Comment (kraftbj) [reply includes/transformer/class-post.php]: Good catch. Wrapped the filter result in `wp_validate_boolean()` so string/int overrides coerce cleanly.

[the remark this answers is not in the document]
Comment (kraftbj) [reply includes/transformer/class-post.php]: Agreed. Added a fallback: when `build_short_form_text()` returns an empty string, we drop back to the long-form teaser composition (title + excerpt + permalink) with the external embed card so we never publish a blank Bluesky post.

File: includes/transformer/class-post.php [hunk 2]
```diff
@@ -49,7 +49,32 @@ class Post extends Base {
...
+		 * Short-form posts publish natively (post body as text, no external
+		 * embed card). Long-form posts use the teaser composition (title +
+		 * excerpt + permalink) with an external card linking back to
+		 * WordPress. The default discriminator mirrors the ActivityPub
+		 * plugin's Post::get_type() logic: untitled posts OR posts with any
+		 * non-empty post_format are short-form.
+		 *
+		 * @param bool     $is_short Whether the post should be treated as short-form.
+		 * @param \WP_Post $post     The post being transformed.
+		 */
```
[the remark this answers is not in the document]
Comment (kraftbj) [reply includes/transformer/class-post.php:65]: Updated the filter docblock to describe all three short-form conditions: post type lacks title support, empty post title, or any non-empty post_format.

File: includes/transformer/class-post.php [hunk 3]
```diff
@@ -209,4 +233,39 @@ public static function upload_thumbnail( int $attachment_id ): ?array {
...
+	 * The post body becomes the Bluesky text directly, with no title
+	 * prefix or trailing permalink. Defensively clamped to 300
+	 * graphemes; a composer UI is expected to enforce the cap before
+	 * publish.
+	 *
+	 * @return string
+	 */
+	private function build_short_form_text(): string {
+		return truncate_text( $this->render_post_content_plain( $this->object ), 300 );
+	}
```
[the remark this answers is not in the document]
Comment (kraftbj) [reply includes/transformer/class-post.php:273]: Aligned the wording with the actual behavior. Updated docblocks in `class-post.php` and `includes/functions.php` to say "300 characters" (mb_strlen code points) rather than graphemes. Grapheme-aware clamping can be a follow-up if we hit real-world issues with combining marks or emoji ZWJ sequences.

File: includes/transformer/class-base.php [hunk 4]
```diff
@@ -133,4 +133,22 @@ protected function get_excerpt( \WP_Post $post, int $word_limit = 30 ): string {
...
+	 *
+	 * @param \WP_Post $post Post object.
+	 * @return string
+	 */
+	protected function render_post_content_plain( \WP_Post $post ): string {
+		$content = \apply_filters( 'the_content', $post->post_content ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound -- Core WordPress filter.
+		$content = \wp_strip_all_tags( $content );
+		$content = \html_entity_decode( $content, ENT_QUOTES, 'UTF-8' );
+
+		return \trim( \preg_replace( '/\s+/', ' ', $content ) );
```
[the remark this answers is not in the document]
Comment (kraftbj) [reply includes/transformer/class-base.php]: Good call. `render_post_content_plain()` now delegates to `Atmosphere\sanitize_text()` after running the `the_content` filter, so the two paths stay in sync by construction.

**Review by pfefferle — APPROVED** [review]
Nice, thanks!
